src/metabase/driver.clj

```diff
@@ -1162,11 +1162,43 @@
 
 (defmulti alter-columns!
   "Alter columns given by `column-definitions` to a table named `table-name`. If the table doesn't exist it will throw an error.
-  Currently we do not currently support changing the the primary key, or take any guidance on how to coerce values."
-  {:added "0.49.0", :arglists '([driver db-id table-name column-definitions])}
+  Currently, we do not currently support changing the primary key, or take any guidance on how to coerce values."
+  {:added "0.49.0"
+   :arglists '([driver db-id table-name column-definitions])
+   :deprecated "0.54.0"}
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
+(defmulti alter-upload-columns!
```

Comment:
How about `alter-table-colums` rather? Seems plausible that this whole data editing project will leverage it as well, for non uploaded tables.

Reply:
Hmm, I did think this might come up. I do think your position is reasonable, but I also think my position (conservative naming / narrower scoping) was reasonable in general (trade-offs). In this instance I defer to your judgement as you have far more awareness of style and wider context! Renamed in https://github.com/metabase/metabase/pull/52607/commits/40ea124477127d5df1990312eabf2f410d4e00f4